sched/clustermembership/cluster.go

```diff
@@ -3,8 +3,37 @@ package cluster_membership
 //go:generate mockgen -source=cluster.go -package=cluster_membership -destination=cluster_mock.go
 
 type Cluster interface {
-	/*
-	 * Returns a Slice of Node Ids in the Cluster
-	 */
+	//
+	// Returns a Slice of Node in the Cluster
+	//
 	Members() []Node
 }
+
+type DynamicCluster interface {
```

Comment:
Yep, it is a picky comment, because I like your code and want it to be the best it can be.

If you think it's a bad suggestion, then skip it, don't block on it, and we can talk about it off the critical path. Or not.

You're right: every implementation will have to implement both sides.

I just want the consumer of the interface to not be confused by extra methods. But maybe there's only one consumer, and it's our core logic, so it's not concerning.